nix/build-from-transcript.nix

```
@@ -35,13 +40,14 @@
 }: let
   compiled = stdenv.mkDerivation {
     # include the ucm version and transcript hash in the derivation name so it is rebuilt if either changes
-    pname = pname + "_ucm-${ucm.version}_${builtins.hashFile "sha256" src}";
+    pname = pname + "_ucm-${ucm.version}_${builtins.hashFile "sha256" transcript}";
```

This is a little awkward because if the src directory changes (a scratch file in it) that could change the output hash as well.

---

I don't think it matters really; either way because whatever files you want are in the derivation, it will be rebuilt when they change.

Or am I missing what this was intending to do?

---

I think that this was guarding against the case that you forgot to update the output hash and so it silently used the old derivation instead of checking the hash. But my memory is fuzzy on why I did this.